### DISCUSSION TITLE
Input css height vs padding to control the height of the input

### DISCUSSION THREAD

**Discussion by veloware** [chakra-ui/chakra-ui#7385] [Q&A]
Hi,

I'm currently restyling our Chakra components to align with a company design system and something I am noticing is how the size of the input is controlled is different from how I have always previously styled inputs, and I'm curious if anyone can shine some light on the different ways to control the height of an input.

In the design system I am following, the height of the input is controlled by a combination of `fontSize`, `lineHeight` and `padding` for example `fontSize: 20px`, `lineHeight: 24px` and `padding-top: 16px`, `padding-bottom: 16px` (would give a total calculated height of `56px`

However when I tried to add an icon using `InputGroup` and `InputLeftElement`, (along with `fontSize: 20px`, `lineHeight: 24px` and `padding-top: 16px`, `padding-bottom: 16px`) the calculated height is 34px, and the icon isn't aligned corrected (horizontally and vertically) and the input itself isn't tall enough

![image](https://user-images.githubusercontent.com/61578546/220651141-62f4f263-3441-49c2-9711-70244ffb64fc.png)

![image](https://user-images.githubusercontent.com/61578546/220651257-866db7d5-7ce0-4b08-8756-ab5820227069.png)

I done some digging around the default styles, and it seems that the height needs to set explicitly, and when I set that, it looks as is expected - 

![image](https://user-images.githubusercontent.com/61578546/220649130-eb5ae66e-46f6-48d3-a60b-1d6543f47832.png)

My question is - have I been styling `input`s incorrectly all my life, and should have been using height rather than a combination of `fontSize`, `lineHeight` and `padding` (Which is also the way the designers have been designing this input)?

**Comment by keithburgie** [comment]
> My question is - have I been styling `input`s incorrectly all my life, and should have been using height rather than a combination of `fontSize`, `lineHeight` and `padding` (Which is also the way the designers have been designing this input)?

If you look at how the MDN Web Docs explain [styling inputs](https://developer.mozilla.org/en-US/docs/Learn/Forms/Styling_web_forms#labels_and_controls), you'll see their suggestion is more like your strategy. So, no! You haven't been doing it wrong.

I went poking around at Chakra's [Input examples](https://chakra-ui.com/docs/components/input) and they do use a fixed height. They also do this with buttons, which is another component I'd size with font styles and padding. I think it works just fine in most cases, though.

A lot of Chakra's compound / composite / multi-part compounds abstract away how props are being passed down. Props attached to `InputGroup` are passed down to children and the input elements are absolutely positioned, which keeps them from centering correctly if you go changing things. A fixed height makes it simpler to position children that way.

Edit: Just want to add that CSS-in-JS makes things like fixed heights more feasible than when it was pure CSS.
